client/src/app/hooks/table-controls/useLocalTableControlDerivedState.ts

```diff
@@ -35,19 +36,38 @@ export const getLocalTableControlDerivedState = <
     >
 ): ITableControlDerivedState<TItem> => {
   const { items, isPaginationEnabled = true } = args;
-  const { filteredItems } = getLocalFilterDerivedState({
-    ...args,
-    items,
-  });
-  const { sortedItems } = getLocalSortDerivedState({
-    ...args,
-    items: filteredItems,
-  });
-  const { currentPageItems } = getLocalPaginationDerivedState({
-    ...args,
-    items: sortedItems,
-  });
+
+  const { filteredItems } = useMemo(
+    () =>
+      getLocalFilterDerivedState({
+        items,
+        filterCategories: args.filterCategories,
+        filterState: args.filterState,
+      }),
+    [items, args.filterCategories, args.filterState]
```

Comment:
Just a note: all inputs are objects/arrays -  the success of caching with `useMemo` depends on how stable the references are (so far they seem stable enough).

Reply:
Yeah can't do much better than that at the moment.  I should read through the table control hook to see if those things are stable at all.